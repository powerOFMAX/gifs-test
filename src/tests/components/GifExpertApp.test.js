import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"

describe('Gif Expert App', () => {
  const title = 'soy un titulo';
  const url = 'https://localhost/asds.jpg';
  const wrapper = shallow(<GifGridItem title={title} url={url}/>)

  test('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('should have a paragraph with a title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  })
  
  test('should have a url', () => {
    const img = wrapper.find('img');
    expect(img.props().src).toBe(url)
    expect(img.props().alt).toBe(title)
  })
  
  test('should have animate__fadeIn', () => {
    const div = wrapper.find('div');
    expect( div.hasClass('animate__fadeIn') ).toBeTruthy();
  })
  
})
