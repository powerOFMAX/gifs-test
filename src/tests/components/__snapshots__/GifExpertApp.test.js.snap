// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif Expert App should match the snapshot 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="soy un titulo"
    src="https://localhost/asds.jpg"
  />
  <p>
    soy un titulo
  </p>
</div>
`;
